File: movinevents/src/main/java/com/movins/events/BlockBase.java
package com.movins.events;

/**
 * 功能：功能块基类
 * Created by movinliao on 2018/4/28.
 */
public class BlockBase extends Dispatcher {
    private Object module;
    private boolean stoped;

    public void init(Object module) {
        this.module = module;
    }

    public void destroy() {
        clear();
        module = null;
    }

    public void start() {
        stoped = false;
        setPaused(false);
    }

    public void stop() {
        setPaused(true);
        stoped = true;
    }

    public boolean isStoped() {
        return stoped;
    }

    public Object getModule() {
        return module;
    }

    public void log(Object obj, String context, Object priority) {
        MethodUtils.apply(module, "log", context, priority);
    }
}

File: src/js/src/module/BlockBase.js
/**
 * 块基类
 * @type {[type]}
 * add by movinliao
 */

var DispImpl = require('../events/DispImpl');

var BlockBase = DispImpl.extend(function () {
  var module = null;
  var stoped = false;

  this.constructor = function (key) {
    this.super(key);
  };
  /**
   * 模块初始化
   * @param api
   *
   */
  this.init = function (mod) {
    module = mod;
  };
  /**
   *模块销毁
   *
   */
  this.destroy = function () {
    this.clear();
    module = null;
  };
  /**
   *停止
   *
   */
  this.stop = function () {
    stoped = true;
    this.setPaused(true);
  };
  /**
   *开始
   *
   */
  this.start = function () {
    stoped = false;
    this.setPaused(false);
  };

  this.stoped = function () {
    return stoped;
  };

  this.module = function () {
    return module;
  };

  this.log = function (context, priority) {
    if ('log' in module) {
      module.log(context, priority);
    }
  };
});

module.exports = BlockBase;
